### DISCUSSION TITLE
docs/web/http-client

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#141] [General]
# docs/web/http-client

GoFrame框架提供的强大HTTP客户端gclient组件，支持HTTP请求的便捷链式操作。客户端支持自定义请求设置及返回对象操作，并详细介绍了超时、Cookie、Header等参数的设置方法。

https://goframe.org/docs/web/http-client

**Comment by IshwarShrivastav** [comment]
通过g.Client()获取了一个文件 如何转为ghttp.UploadFile？
ghttp.UploadFile只能通过前端传过来吗？
后端自己请求文件 能不能转为ghttp.UploadFile？
或者有一个文件能塞到ghttp.UploadFile中？

**Reply by gqcn** [reply]
> 通过g.Client()获取了一个文件 如何转为ghttp.UploadFile？ ghttp.UploadFile只能通过前端传过来吗？ 后端自己请求文件 能不能转为ghttp.UploadFile？ 或者有一个文件能塞到ghttp.UploadFile中？

你好，可以参考框架中的相关创建方法：https://github.com/gogf/gf/blob/6aa5c2b2ef7683c63d3f9b9521e58899a52e4446/net/ghttp/ghttp_request_param_file.go#L129

但其实框架如果能提供根据文件创建`ghttp.UploadFile`对象的方法就更好了，感兴趣可以参与共建，也可以提交`issue`记录，社区后续实现该方法。

**Comment by github-oldman** [comment]
g.Client().Timeout(40*time.Second)，时长超过30秒就没用了，设置的超时时间不生效，只能30秒，有什么方法可以设置吗？
